pkg/config/autodiscover/autodiscover_targets.go

```diff
@@ -66,6 +70,54 @@ func FindTestTarget(labels []configsections.Label, target *configsections.TestTa
 	}
 
 	target.DeploymentsUnderTest = append(target.DeploymentsUnderTest, FindTestDeployments(labels, target, namespace)...)
+	target.Nodes = GetNodesList()
+}
+
+// GetNodesList Function that return a list of node and what is the type of them.
+func GetNodesList() (nodes map[string]configsections.Node) {
+	nodes = make(map[string]configsections.Node)
+	var nodeNames []string
+	var types []string
+	context := common.GetContext()
+	tester := nodenames.NewNodeNames(common.DefaultTimeout, map[string]*string{"node-role.kubernetes.io/master": nil})
+	test, _ := tnf.NewTest(context.GetExpecter(), tester, []reel.Handler{tester}, context.GetErrorChannel())
+	_, err := test.Run()
+	if err != nil {
+		log.Error("Unable to get node list ", ". Error: ", err)
+	} else {
+		nodeNames = tester.GetNodeNames()
+		for i := range nodeNames {
+			node := configsections.Node{
+				Name: nodeNames[i],
+				Type: append(types, "master"),
+			}
+			nodes[nodeNames[i]] = node
+		}
+	}
+	context = common.GetContext()
+	tester = nodenames.NewNodeNames(common.DefaultTimeout, map[string]*string{"node-role.kubernetes.io/worker": nil})
+	test, _ = tnf.NewTest(context.GetExpecter(), tester, []reel.Handler{tester}, context.GetErrorChannel())
+	_, err = test.Run()
+	if err != nil {
+		log.Error("Unable to get node list ", ". Error: ", err)
```

Comment:
Same here, but the nodes variable should be emptied before returning.

Reply:
i think there is no need to return here ,,, if we got here it mean that we have master node and we wanto go to the final return that we have in this function

Reply:
You're right. I just noticed that with a return here you can save one indentation because the "else" would not be needed for that part.